src/commands/build.yml

```diff
@@ -1,5 +1,18 @@
 description: Builds source code using go modules, doesn't modify the go mod file
+parameters:
+  github-username:
+    description: username of user with access to private github repositories
+    type: env_var_name
+    default: GITHUB_USERNAME
+  github-token:
+    description: token to for the user with access to private github repositories
+    type: env_var_name
+    default: GITHUB_TOKEN
 steps:
   - run:
       name: Build source code
-      command: go build -mod=readonly -v ./...
+      command: |
+        git config --global --unset url."ssh://git@github.com".insteadOf
+        export GOPRIVATE="github.com/Financial-Times"
+        git config --global url."https://${<<parameters.github-username>>}:${<<parameters.github-token>>}@github.com".insteadOf "https://github.com"
```

Comment:
If I'm reading this correctly, this change expects always to have `GITHUB_USERNAME` and `GITHUB_TOKEN` environment variables. Are those variables setup by default by CircleCI? Or are we suppose to set them up on every project?

Reply:
no, they are not set by default so you have to add them to every pipeline that needs them. 
If they are missing and the project does not have private dependencies - it would still work.